### PR TITLE
Dev get peer ID

### PR DESCRIPTION
**Pull request by otaviolcs3**

### CODE REVIEW

File: spiffetls/tests/check_mode.c [hunk 1]
```diff
@@ -141,6 +147,95 @@ START_TEST(test_spiffetls_MTLSWebClientWithRawConfig)
...
+    ck_assert_uint_eq(mode->mode, TLS_SERVER_MODE);
+    ck_assert_ptr_eq(mode->source, source);
+
+    spiffetls_ListenMode_Free(mode);
+}
+END_TEST
+
+START_TEST(test_spiffetls_TLSServerWithRawConfig)
+{
+    x509svid_Source *svid = NULL;
```
Comment (rodrigolc) [inline spiffetls/tests/check_mode.c:178]: Shouldn't we use a non-null pointer for this? null pointers should give us an error

File: spiffetls/tests/check_peerid.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+#include "spiffetls/src/peerid.h"
+#include "spiffetls/tlsconfig/src/config.h"
+#include <check.h>
+#include <stdio.h>
+#include <unistd.h>
+
+START_TEST(test_spiffetls_PeerIDFromConn)
+{
+    /* Hot path */
+    system("./tls_server resources/good-leaf-only.pem "
```
Comment (rodrigolc) [inline spiffetls/tests/check_peerid.c:12]: Remember other tests should run in parallel and not fail.
This fails to acquire the port if another "tls_server" is also running, including for other tests.
Changing the port used should suffice, or even better a randomly chosen passed from the test to the "tls_server" program via args.

File: spiffetls/src/mode.c [hunk 3]
```diff
@@ -125,3 +125,85 @@ void spiffetls_DialMode_Free(spiffetls_DialMode *mode)
...
+    return mode;
+}
+
+spiffetls_ListenMode *
+spiffetls_TLSServerWithSource(workloadapi_X509Source *source)
+{
+    spiffetls_ListenMode *mode = malloc(sizeof *mode);
+    memset(mode, 0, sizeof *mode);
+    mode->mode = TLS_SERVER_MODE;
+    mode->source = source;
```
Comment (rodrigolc) [inline spiffetls/src/mode.c:144]: Null checking?

File: spiffetls/src/mode.c [hunk 4]
```diff
@@ -125,3 +125,85 @@ void spiffetls_DialMode_Free(spiffetls_DialMode *mode)
...
+
+    return mode;
+}
+
+spiffetls_ListenMode *spiffetls_TLSServerWithRawConfig(x509svid_Source *svid)
+{
+    spiffetls_ListenMode *mode = malloc(sizeof *mode);
```
Comment (rodrigolc) [inline spiffetls/src/mode.c:151]: same

File: spiffetls/src/mode.c [hunk 5]
```diff
@@ -125,3 +125,85 @@ void spiffetls_DialMode_Free(spiffetls_DialMode *mode)
...
+    mode->mode = TLS_SERVER_MODE;
+    mode->unneeded_source = true;
+    mode->svid = svid;
+
+    return mode;
+}
+
+spiffetls_ListenMode *spiffetls_MTLSServer(tlsconfig_Authorizer *authorizer)
+{
+    spiffetls_ListenMode *mode = malloc(sizeof *mode);
```
Comment (rodrigolc) [inline spiffetls/src/mode.c:162]: same

File: spiffetls/src/mode.c [hunk 6]
```diff
@@ -125,3 +125,85 @@ void spiffetls_DialMode_Free(spiffetls_DialMode *mode)
...
+
+    return mode;
+}
+
+spiffetls_ListenMode *
+spiffetls_MTLSServerWithSource(tlsconfig_Authorizer *authorizer,
+                               workloadapi_X509Source *source)
+{
+    spiffetls_ListenMode *mode = malloc(sizeof *mode);
+    memset(mode, 0, sizeof *mode);
```
Comment (rodrigolc) [inline spiffetls/src/mode.c:175]: same

File: spiffetls/src/mode.c [hunk 7]
```diff
@@ -125,3 +125,85 @@ void spiffetls_DialMode_Free(spiffetls_DialMode *mode)
...
+spiffetls_ListenMode *
+spiffetls_MTLSServerWithRawConfig(tlsconfig_Authorizer *authorizer,
+                                  x509svid_Source *svid,
+                                  x509bundle_Source *bundle)
+{
+    spiffetls_ListenMode *mode = malloc(sizeof *mode);
+    memset(mode, 0, sizeof *mode);
+    mode->mode = MTLS_SERVER_MODE;
+    mode->unneeded_source = true;
+    mode->authorizer = authorizer;
```
Comment (rodrigolc) [inline spiffetls/src/mode.c:192]: same, but with 3 pointers

**Review by rodrigolc — CHANGES_REQUESTED** [review]

File: spiffetls/tests/check_dial.c [hunk 8]
```diff
@@ -17,7 +17,7 @@ START_TEST(test_spiffetls_DialWithMode)
     spiffetls_DialMode *mode
         = spiffetls_TLSClientWithRawConfig(authorizer, bundle_src);
 
-    spiffetls_dialConfig config = { .base_TLS_conf = NULL, .dialer_fd = 0 };
+    spiffetls_dialConfig config = { .base_TLS_conf = NULL, .dialer_fd = -1 };
```
[the remark this answers is not in the document]
Comment (otaviolcs3) [reply spiffetls/tests/check_dial.c:20]: Means invalid dialer

**Review by rodrigolc — APPROVED** [review]

**Review by willallves — APPROVED** [review]

**Pull request merged**